dashboard/client/src/components/NewEventTable.tsx

```diff
@@ -0,0 +1,335 @@
+};
+
+const transformFiltersToParams = (filters: Filters) => {
+  const params = new URLSearchParams();
+
+  // Handling special cases for entityName and entityId
+  if (filters.entityName && filters.entityId) {
+    appendToParams(params, filters.entityName, filters.entityId);
+  }
+
+  // Handling general cases, for key like "count", "sourceType", "severityLevel"
+  Object.entries(filters).forEach(([key, value]) => {
+    if (key !== "entityName" && key !== "entityId") {
+      appendToParams(params, key, value as string | string[]);
+    }
+  });
+
+  return params.toString();
+};
+
+const useStyles = makeStyles((theme) => ({
+  overflowCell: {
+    display: "block",
+    margin: "auto",
+    maxWidth: 360,
+    textOverflow: "ellipsis",
+    overflow: "hidden",
+    whiteSpace: "nowrap",
+  },
+  filterContainer: {
+    display: "flex",
+    alignItems: "center",
+  },
+  search: {
+    margin: theme.spacing(1),
+    display: "inline-block",
+    fontSize: 12,
+    lineHeight: "46px",
+    height: 56,
+  },
+
+  tableContainer: {
+    overflowX: "scroll",
+  },
+
+  helpInfo: {
+    marginLeft: theme.spacing(1),
+  },
+  message: {
+    maxWidth: 200,
+  },
+  pagination: {
+    marginTop: theme.spacing(3),
+  },
+}));
+
+const SOURCE_TYPE_OPTIONS = [
+  "COMMON",
+  "CORE_WORKER",
+  "GCS",
+  "RAYLET",
+  "JOBS",
+  "SERVE",
+  "CLUSTER_LIFECYCLE",
+  "AUTOSCALER",
+];
+
+const SEVERITY_LEVEL_OPTIONS = ["INFO", "DEBUG", "WARNING", "ERROR", "TRACING"];
+
+const COLUMNS = [
+  { label: "Severity", align: "center" },
+  { label: "Timestamp", align: "center" },
+  { label: "Source", align: "center" },
+  { label: "Custom Fields", align: "left" },
+  { label: "Message", align: "left" },
+];
+
+const useEventTable = (props: EventTableProps) => {
+  const { defaultSeverityLevels, entityName, entityId } = props;
+  const [loading, setLoading] = useState(true);
+  const { changeFilter: _changeFilter, filterFunc } = useFilter();
+  const [filters, _setFilters] = useState<Filters>({
+    sourceType: [],
+    severityLevel: defaultSeverityLevels || [],
+    entityName, // We used two fields(entityName, entityId) because we will support select entityName by dropdown and input entityId by TextField in the future.
+    entityId, // id or *
+  });
+
+  const [events, setEvents] = useState<Event[]>([]);
+
+  const [pagination, setPagination] = useState({
+    pageNo: 1, // first page is PageNo 1
+    pageSize: 10,
+  });
+
+  const changePage = (key: string, value: number) => {
+    setPagination({ ...pagination, [key]: value });
+  };
+
+  const changeFilter: typeof _changeFilter = (...params) => {
+    _changeFilter(...params);
+    setPagination({
+      ...pagination,
+      pageNo: 1,
+    });
+  };
+
+  const setFilters: typeof _setFilters = (...params) => {
+    _setFilters(...params);
+    setPagination({
+      ...pagination,
+      pageNo: 1,
+    });
+  };
+  const { pageNo } = pagination;
+
+  const params = transformFiltersToParams(filters);
```

Comment:
instead of calculating the params here and passing it all the way to the `getEvent` call.

Can we pass the filters to `getEvent` and then have `getEvent` do the transformation?

It's better because then we have a higher-level interface for `getEvent` (accepts filters instead of a raw string)

and also will allow us to avoid doing this transformation logic with every single render.

Reply:
Sounds good!